.github/workflows/test.yml

```diff
@@ -26,7 +26,6 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools
           pip install -e .[ci]
```

Comment:
The advantage of pep 517 is users won't have to explicitly install setuptools when on python 3.12. Pip can take care of this for them since it knows what build system we are using in advance.